### ISSUE TITLE
Back Up and Restore For a limited period 

### ISSUE THREAD

**Issue by mohanagandlla** [prometheus-operator/prometheus-operator#3632]
I have created the snapshot by API it is created and restore also done my question is....
The snap short at every time is creating from the very beginning of the data how to take the snapshot with a limitation like day bases week bases and month bases?

**Comment by nrchakradhar** [comment]
@mohanagandlla What is the role of the operator for snapshot. Snapshot API is admin REST API of Prometheus. Are you expecting some additional functionality in operator?

**Comment by mohanagandlla** [comment]
Yes the role is Admin and what I am expecting here is the snapshots can be able to take on different time sequences 
ex:
If I take the first snapshot today and again I am going to take the second snapshot on next day but here the major point is at the second day the snapshot contains the first data too I need to collect the only present day's data and make sure I need to maintain the retention time is 10 days.

**Comment by nrchakradhar** [comment]
Snapshot is a feature of prometheus and from prometheus documentation, it will be whatever is present in the TSDB data directory.
https://prometheus.io/docs/prometheus/latest/querying/api/#snapshot
For which timelines the data is present in TSDB data directory depends on the parameters configured for retention(size, time). So what you want to achieve may be possible in an indirect way.
It might be possible to add a feature to the operator to trigger a backup at some regular intervals. This however can also be achieved in different ways and would like to hear from others.

**Comment by mohanagandlla** [comment]
ok fine So what I have done here is I am maintaining the retention as X days so I have planed a corn job for same X days to generate the snapshot and that will copy to storage driver finally the snapshot will delete inside the pod. By this one we will have the data period of X days after X days a new data will come and we can save the pod memory too.

I am facing another issue here the above one is working fine but I am maintaining the 2 replicas both are having different PV so when I took a snapshot it is performing on the first replica who to take on 2nd replica?
